### DISCUSSION TITLE
Using state in embeds

### DISCUSSION THREAD

**Discussion by gemathus** [outline/outline#3584] [General]
Hi!

I was wondering if there is a way, or how would you go about using state in an Embed component. I would like to use state on an Embed component.

**Comment by Limezy** [comment]
Hi, I didn't understand your question ?

**Reply by gemathus** [reply]
Hi @Limezy ! Sorry, I wasn't clear at all. 

Let's look for example at the YouTube Embed Component, and say we want this component to have its own state. Normally, we would initialize state in the constructor

````ts
export default class YouTube extends React.Component<Props> {
  static ENABLED = [URL_REGEX];
  constructor(props: Props) {
    this.state = { foo: "bar" }
  }
//...
````
We then use this state somewhere in the component:
````ts
console.log(this.state.foo)
````
Although this works, I have a feeling that it's not te right call architecture-wise. Also, VS Code highlights this usage as en error:
`Property 'foo' does not exist on type 'Readonly<{}>'`
